File: src/main/java/com/example/sns/dto/request/PostCreateRequest.java
package com.example.sns.dto.request;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;

/**
 * 게시글 작성 요청.
 * 프론트엔드 호환: imageUrl, category 필드 추가.
 */
public record PostCreateRequest(
        @NotBlank(message = "제목은 필수입니다.")
        @Size(max = 200)
        String title,

        @NotBlank(message = "내용은 필수입니다.")
        String content,

        Double latitude,
        Double longitude,
        Long pinId,
        String imageUrl,
        String category
) {
}

File: mobile/lib/domain/models/post_create_request.dart
/// doc/API_SPEC.md 4.3 — PostCreateRequest
class PostCreateRequest {
  const PostCreateRequest({
    required this.title,
    required this.content,
    this.latitude,
    this.longitude,
    this.pinId,
  });

  final String title;
  final String content;
  final double? latitude;
  final double? longitude;
  final int? pinId;

  Map<String, dynamic> toJson() => {
        'title': title,
        'content': content,
        if (latitude != null) 'latitude': latitude,
        if (longitude != null) 'longitude': longitude,
        if (pinId != null) 'pinId': pinId,
      };
}
